astropy/io/fits/src/compressionmodule.c

```diff
@@ -1059,9 +1059,36 @@ PyObject* compression_decompress_hdu(PyObject* self, PyObject* args)
         return NULL;
     }
 
-    bitpix_to_datatypes(fileptr->Fptr->zbitpix, &datatype, &npdatatype);
-    if (PyErr_Occurred()) {
-        return NULL;
+    fits_get_img_equivtype(fileptr, &datatype, &status);
+    if (status != 0) {
```

Comment:
Is it possible that this should be `> 0` instead of `!= 0`? That's what the `ffgiet` function checks.

Reply:
That could also explain the `unknown error` problems.